ingest/src/main/java/com/microsoft/azure/kusto/ingest/StreamingIngestClient.java

```diff
@@ -147,34 +146,6 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
         return new IngestionStatusResult(ingestionStatus);
     }
 
-    private String getFormat(IngestionProperties ingestionProperties) {
-        String format = ingestionProperties.getDataFormat();
-        if (format == null) {
-            return "csv";
-        }
-        return format;
-    }
-
-    private String getMappingReference(IngestionProperties ingestionProperties, String format) throws IngestionClientException {
-        IngestionMapping ingestionMapping = ingestionProperties.getIngestionMapping();
-        String mappingReference = ingestionMapping.getIngestionMappingReference();
-        IngestionMapping.IngestionMappingKind ingestionMappingKind = ingestionMapping.getIngestionMappingKind();
-        if (IngestionMapping.mappingRequiredFormats.contains(format)) {
-            String message = null;
-            if (!format.equalsIgnoreCase(ingestionMappingKind.name())) {
```

Comment:
This is not identical to the checks in `IngestionProperties.validate()`. There, `IngestionMappingKind` can be `null` or `unknown` and not fail this. The question is what is correct - both in all cases, or whether `StreamingClient` is a special case?

@ArielYehezkely ?

Reply:
How would unknown fail this?

Reply:
See in the `IngestionProperties.validate()` method:
`if (dataFormat != null && ingestionMappingKind != null && ingestionMappingKind != IngestionMapping.IngestionMappingKind.unknown && !dataFormat.getIngestionMappingKind().equals(ingestionMappingKind))`

Reply:
You added this line in this pr :)

But still my point stands - unknown wouldn't fail either

Reply:
Right, my point is that they're not identical, so I'm trying to identify which is the correct approach.

Reply:
I think unknown shouldn't exist, or should always be an error when the user passes it.

I remember hearing that txt and raw can have csv mapping, and if not they shouldn't have mapping at all.

I think mapping kind should always be provided, so null should fail it too.

Reply:
Nevermind, I aligned this check to how it originally was. I was missing a `&&  dataFormat.isMappingRequired()`, though they are effectively identical as we already have a check a few lines up whether a mapping is defined if mappings are required.
The code just makes a bit more sense now :-)